examples/game_data/main.rs

```diff
@@ -0,0 +1,583 @@
+    white: Material,
+    logo: Material,
+    font: FontHandle,
+}
+
+fn load_assets(world: &World) -> (ProgressCounter, Assets) {
+    let mut progress = ProgressCounter::default();
+    let mesh_storage = world.read_resource();
+    let tex_storage = world.read_resource();
+    let font_storage = world.read_resource();
+    let mat_defaults = world.read_resource::<MaterialDefaults>();
+    let loader = world.read_resource::<Loader>();
+
+    let red = loader.load_from_data([1.0, 0.0, 0.0, 1.0].into(), &mut progress, &tex_storage);
+    let red = Material {
+        albedo: red,
+        ..mat_defaults.0.clone()
+    };
+
+    let white = loader.load_from_data([1.0, 1.0, 1.0, 1.0].into(), &mut progress, &tex_storage);
+    let white = Material {
+        albedo: white,
+        ..mat_defaults.0.clone()
+    };
+
+    let logo = Material {
+        albedo: loader.load(
+            "texture/logo.png",
+            PngFormat,
+            Default::default(),
+            &mut progress,
+            &tex_storage,
+        ),
+        ..mat_defaults.0.clone()
+    };
+
+    let cube = loader.load("mesh/cube.obj", ObjFormat, (), &mut progress, &mesh_storage);
+    let cone = loader.load("mesh/cone.obj", ObjFormat, (), &mut progress, &mesh_storage);
+    let lid = loader.load("mesh/lid.obj", ObjFormat, (), &mut progress, &mesh_storage);
+    let teapot = loader.load(
+        "mesh/teapot.obj",
+        ObjFormat,
+        (),
+        &mut progress,
+        &mesh_storage,
+    );
+    let rectangle = loader.load(
+        "mesh/rectangle.obj",
+        ObjFormat,
+        (),
+        &mut progress,
+        &mesh_storage,
+    );
+    let font = loader.load(
+        "font/square.ttf",
+        TtfFormat,
+        (),
+        &mut progress,
+        &font_storage,
+    );
+
+    (
+        progress,
+        Assets {
+            cube,
+            cone,
+            lid,
+            rectangle,
+            teapot,
+            red,
+            white,
+            logo,
+            font,
+        },
+    )
+}
+
+fn main() {
+    if let Err(error) = run() {
+        eprintln!("Could not run the example!");
+        eprintln!("{}", error);
+        ::std::process::exit(1);
+    }
+}
+
+/// Wrapper around the main, so we can return errors easily.
+fn run() -> Result<(), Error> {
+    // Add our meshes directory to the asset loader.
+    let resources_directory = format!("{}/examples/assets", env!("CARGO_MANIFEST_DIR"));
+
+    let display_config_path = format!(
+        "{}/examples/renderable/resources/display_config.ron",
+        env!("CARGO_MANIFEST_DIR")
+    );
+
+    let display_config = DisplayConfig::load(display_config_path);
+    let pipeline_builder = Pipeline::build().with_stage(
+        Stage::with_backbuffer()
+            .clear_target([0.0, 0.0, 0.0, 1.0], 1.0)
+            .with_pass(DrawShaded::<PosNormTex>::new())
+            .with_pass(DrawUi::new()),
+    );
+    let game_data = GameDataBuilder::default()
```

Comment:
Games don't only have a base and a running state. There should be a way to provide multiple systems that are specific to each of the states. Those systems should be added during the state start function, to allow for better logic locality.

Reply:
Ofc not, but I'm not about to write a whole game in a single example, and this example is for demonstrating a custom game data implementation, not teach people how to structure a game code wise.